code/_onclick/ventcrawl.dm

```diff
@@ -0,0 +1,46 @@
+proc/isvent(var/atom/A)
+	if(istype(A,/obj/machinery/atmospherics/unary/vent_pump))
+		return 1
+	return 0
+
+/mob/living/carbon/slime/AltClickOn(var/atom/A)
```

Comment:
You repeat this code a lot.
I suggest adding it in at the mob level and then defining a var for which mobs can ventcrawl, or something like that.
Consider adding it to ghosts as well.